src/service/data_products/heatmap.py

```diff
@@ -233,6 +268,49 @@ async def get_cell(
         )
         return heatmap_models.CellDetail(**remove_collection_keys(doc))
 
+    def _append_col(
+            self,
+            columns: list[col_models.AttributesColumn],
+            col_list: list[str],
+            column_type: col_models.ColumnType,
+            filter_strategy: col_models.FilterStrategy) -> None:
+        # create AttributesColumn objects from a list of column names and add them, in place, to the given columns list
+        for col_name in col_list:
+            columns.append(col_models.AttributesColumn(
+                key=col_name,
+                type=column_type,
+                filter_strategy=filter_strategy,
+            ))
+
+    async def _get_heatmap_columns(
+            self,
+            storage: ArangoStorage,
+            coll_id: str,
+            load_ver: str,
+            load_ver_override: bool
+    ) -> list[col_models.AttributesColumn]:
+        # Retrieve a list of AttributesColumn objects derived from the ColumnInformation objects within HeatMapMeta.
+        # Additionally, include columns that exist in the heatmap row data but are not present in HeatMapMeta.
+
+        doc = await get_collection_singleton_from_db(
+            storage, self._colname_meta, coll_id, load_ver, bool(load_ver_override))
+        column_meta = heatmap_models.HeatMapMeta(**remove_collection_keys(doc))
```

Comment:
Hmm, this is the same as in the get_meta method which makes me thing it should be made into a method

Reply:
👍

Reply:
:+1: